src/FirmwarePlugin/APM/APMFirmwarePlugin.cc

```diff
@@ -161,13 +246,80 @@ void APMFirmwarePlugin::adjustMavlinkMessage(mavlink_message_t* message)
                 // Already in param_float
                 break;
             default:
-                qCritical() << "Invalid/Unsupported data type used in parameter:" << paramSet.param_type;
+                qCCritical(APMFirmwarePluginLog) << "Invalid/Unsupported data type used in parameter:" << paramSet.param_type;
         }
         
         mavlink_msg_param_set_encode(message->sysid, message->compid, message, &paramSet);
     }
 
-    // FIXME: Need to implement mavlink message severity adjustment
+    if (message->msgid == MAVLINK_MSG_ID_STATUSTEXT)
+    {
+        QByteArray b;
+        b.resize(MAVLINK_MSG_STATUSTEXT_FIELD_TEXT_LEN+1);
+        mavlink_msg_statustext_get_text(message, b.data());
+        // Ensure NUL-termination
+        b[b.length()-1] = '\0';
+        QString text = QString(b);
+        qCDebug(APMFirmwarePluginLog) << text;
```

Comment:
Can you put this part inside the if as well? Doesn't need to happen on every message does it? You'll only get the log output if you need to parse a version but I think that makes sense.